column/delivery/http/handler.go

```
@@ -0,0 +1,133 @@
+package http
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/PECHIVKO/task-manager/column"
+	"github.com/go-chi/chi"
+)
+
+type Handler struct {
+	useCase column.UseCase
+}
+
+func NewHandler(useCase column.UseCase) *Handler {
+	return &Handler{
+		useCase: useCase,
+	}
+}
+
+type createInput struct {
+	Name      string `json:"column_name"`
+	ProjectID int    `json:"project_id"`
+}
+
+func (h *Handler) Create(w http.ResponseWriter, r *http.Request) {
+	input := new(createInput)
+	err := json.NewDecoder(r.Body).Decode(&input)
+	if err != nil {
+		log.Println(err)
+		respondWithError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	err = h.useCase.CreateColumn(r.Context(), input.Name, input.ProjectID)
+	if err != nil {
+		log.Println(err)
+		respondWithError(w, http.StatusInternalServerError, err.Error())
+	} else {
+		respondwithJSON(w, http.StatusCreated, map[string]string{"message": "Column successfully created"})
+	}
+}
+
+type updateNameInput struct {
+	ID   int    `json:"column_id"`
+	Name string `json:"column_name"`
+}
+
+func (h *Handler) UpdateName(w http.ResponseWriter, r *http.Request) {
+	input := new(updateNameInput)
+	err := json.NewDecoder(r.Body).Decode(&input)
+	if err != nil {
+		log.Println(err)
+		respondWithError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	input.ID, _ = strconv.Atoi(chi.URLParam(r, "id"))
+
+	err = h.useCase.UpdateColumnName(r.Context(), input.Name, input.ID)
+	if err != nil {
+		log.Println(err)
+		respondWithError(w, http.StatusInternalServerError, err.Error())
+	} else {
+		respondwithJSON(w, http.StatusOK, map[string]string{"message": "Column successfully updated"})
+	}
+}
+
+func (h *Handler) Move(w http.ResponseWriter, r *http.Request) {
+	id, _ := strconv.Atoi(chi.URLParam(r, "id"))
```

do not ignore errors